trin-core/src/portalnet/find/query_info.rs

```diff
@@ -18,6 +19,8 @@ pub struct QueryInfo<TContentKey> {
 
     /// Temporary ENRs used when trying to reach nodes.
     pub untrusted_enrs: SmallVec<[Enr; 16]>,
+
+    pub trace: Option<QueryTrace>,
```

Comment:
I'm not sure that the trace belongs in `QueryInfo`, because the trace is specific to content queries, and `QueryInfo` exists for both node and content queries.

Reply:
I think that having traces of node queries could be good too, so I'll keep it here for now and move it elsewhere if we decide to never create traces of node queries.